### DISCUSSION TITLE
Add progress bar to reading in a file?

### DISCUSSION THREAD

**Discussion by MicahGale** [idaholab/MontePy#346] [Ideas]
Many whole core models take over 40 seconds to read in. Would a progress bar help with tolerating this pause?

No matter what the bar needs to be suppressible. The question is should it default off or on?

**Comment by MicahGale** [comment]
All of these solutions I'm not a fan of because they require another PYPI package with little added value: https://builtin.com/software-engineering-perspectives/python-progress-bar

Though manually implementing a progress counter shouldn't be too hard.... (famous last words).

I think this will need some benchmarking from #96, because it seems like most of the slow down is actually in the `update_pointers` stage.

**Comment by MicahGale** [comment]
In GitLab by @tjlaboss on Mar 14, 2023, 16:26

Quick-and-dirty progress bar

```python
import time

def bar():
    for i in range(10 + 1):
        string = '[' + '='*i + '.'*(10-i) + ']'
        if i > 0:
            print('\b'*len(string), end='')
        print(string, end="", flush=True)
        time.sleep(.1)
    print()
```

**Comment by MicahGale** [comment]
In GitLab by @tjlaboss on Mar 14, 2023, 16:32

The progress bar default could be different if between a script and a live interpreter. 

An argument like `quiet=True` or `silent=True` is common. If the argument is `silent`, it should suppress warnings too.

**Comment by MicahGale** [comment]
Thoughts on how to implement this:

1. benchmark loading and figure out which stages take more time, parsing, or post-parsing problem setup
2. Read number of lines in file quickly
3. Estimate parsing progress based on line number out of total
4. If 90% is parsing once EOF is reached loading bar is 90%, etc.

**Comment by SterlingButters** [comment]
@MicahGale See [rich](https://github.com/Textualize/rich)'s progress bar - it's prettier and likely easy to implement
